js_modules/dagster-ui/packages/ui-core/src/search/useGlobalSearch.tsx

```diff
@@ -164,15 +170,14 @@ const primaryDataToSearchResults = (input: {data?: SearchPrimaryQuery}) => {
             type: SearchResultType.PartitionSet,
           }));
 
-        return [
-          ...inner,
-          ...allAssetGroups,
-          ...allPipelinesAndJobs,
-          ...allSchedules,
-          ...allSensors,
-          ...allPartitionSets,
-          ...allResources,
-        ];
+        inner.push(...allAssetGroups);
+        inner.push(...allPipelinesAndJobs);
+        inner.push(...allSchedules);
+        inner.push(...allSensors);
+        inner.push(...allPartitionSets);
+        inner.push(...allResources);
+
+        return inner;
```

Comment:
Why?

Reply:
performance, there's no reason to recreate the `inner` array on every iteration